### ISSUE TITLE
Incompatibility with Guava 32.1.1

### ISSUE THREAD

**Issue by markusheiden** [GoogleCloudPlatform/spring-cloud-gcp#2067]
**Describe the bug**
When updating the Guava version from 32.0.1-jre to 32.1.1-jre in my project, Gradle complains about a `conflict on capability 'com.google.guava:listenablefuture:1.0'`.

Gradle output:
```
> Could not resolve all files for configuration ':gcp:compileClasspath'.
   > Could not resolve com.google.guava:guava:32.1.1-jre.
     Required by:
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.api:api-common:2.14.0
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.api:gax:2.31.0
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.cloud:google-cloud-core:2.21.0
         project :gcp > com.google.cloud:spring-cloud-gcp-starter-logging:4.6.0 > com.google.cloud:spring-cloud-gcp-logging:4.6.0 > com.google.cloud:google-cloud-logging-logback:0.130.18-alpha
      > Module 'com.google.guava:guava' has been rejected:
           Cannot select module with conflict on capability 'com.google.guava:listenablefuture:1.0' also provided by [com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava(compile)]
   > Could not resolve com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava.
     Required by:
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.api.grpc:proto-google-cloud-translate-v3beta1:0.103.0
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.api.grpc:proto-google-cloud-translate-v3:2.21.0
         project :gcp > com.google.cloud:spring-cloud-gcp-starter-logging:4.6.0 > com.google.cloud:spring-cloud-gcp-logging:4.6.0 > com.google.cloud:google-cloud-logging-logback:0.130.18-alpha
      > Module 'com.google.guava:listenablefuture' has been rejected:
           Cannot select module with conflict on capability 'com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava' also provided by [com.google.guava:guava:32.1.1-jre(jreApiElements)]
   > Could not resolve com.google.guava:guava:31.1-jre.
     Required by:
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.protobuf:protobuf-java-util:3.23.2
      > Module 'com.google.guava:guava' has been rejected:
           Cannot select module with conflict on capability 'com.google.guava:listenablefuture:1.0' also provided by [com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava(compile)]
   > Could not resolve com.google.guava:guava:29.0-android.
     Required by:
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > io.opencensus:opencensus-contrib-http-util:0.31.1
      > Module 'com.google.guava:guava' has been rejected:
           Cannot select module with conflict on capability 'com.google.guava:listenablefuture:1.0' also provided by [com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava(compile)]
   > Could not resolve com.google.guava:guava:32.0.0-android.
     Required by:
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.auth:google-auth-library-oauth2-http:1.19.0
      > Module 'com.google.guava:guava' has been rejected:
           Cannot select module with conflict on capability 'com.google.guava:listenablefuture:1.0' also provided by [com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava(compile)]
   > Could not resolve com.google.guava:guava:30.1.1-android.
     Required by:
         project :gcp > com.google.cloud:google-cloud-translate:2.21.0 > com.google.http-client:google-http-client:1.43.3
      > Module 'com.google.guava:guava' has been rejected:
           Cannot select module with conflict on capability 'com.google.guava:listenablefuture:1.0' also provided by [com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava(compile)]
```

Somehow the strange `com.google.guava:listenablefuture:9999.0-empty-to-avoid-conflict-with-guava` workaround seems not to work.

**Comment by markusheiden** [comment]
Yes, but we use Gradle:

```
dependencyManagement {
    imports {
        mavenBom "com.google.cloud:spring-cloud-gcp-dependencies:4.6.0"
        mavenBom "org.springframework.cloud:spring-cloud-dependencies:2022.0.3"
        mavenBom "org.springframework.boot:spring-boot-dependencies:3.1.2"
    }

    dependencies {
        ...
        dependency "com.google.guava:guava:32.0.1-jre"
        ...
    }
}
```

**Comment by meltsufin** [comment]
@markusheiden There is a version of Guava 32.0.1 that we are currently compatible with. Why are you trying to override the version of Guava in your application?

@suztomo Do you know why we are not upgrading Guava to latest in the client libraries? The latest version is 32.1.1.

**Comment by suztomo** [comment]
There're workaround in Guava's the dependency conflicts https://github.com/google/guava/releases/tag/v32.1.0

> the fixed Gradle metadata in 32.1.1 may still require action on your part.

To Mike's question, why the Guava version
https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/a990b8528c587b9c653740463893b4503f9b25c9/spring-cloud-gcp-dependencies/pom.xml#L39 is not updated, I think it's because https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/renovate.json#L16

```
		"major": {
			"enabled": false
```

**Comment by markusheiden** [comment]
@meltsufin We use Guava ourselves and strive to update ASAP to avoid missing any bug fixes.
